File: programming_projects/python/swap.c
#include <stdio.h>

void swap(int* a, int* b){
    int temp = *a;
    *a = *b;
    *b = temp;
}

int main(void){
    int x,y;
    x = 1;
    y = 2;
    swap(&x,&y);
    printf("X contains %d, Y contains %d", x, y);
}

File: programming_projects/python/swap.py
x = 1
y = 2
def swap( a, b):
    print ("a contain the value {} and b contains the value {}".format(a,b))
    temp = a
    a = b
    b = temp
    print ("a contain the value {} and b contains the value {}".format(a,b))
swap( x, y)
print("x contains the value {} and y contains the value{}".format(x, y))